### PR TITLE
QgsDataSourceURI doxygen improvements

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

File: python/core/auto_generated/qgsdatasourceuri.sip.in [hunk 1]
```diff
@@ -139,144 +145,166 @@ Sets database
...
     void setPassword( const QString &password );
 %Docstring
-Sets password
+Sets the ``password`` for the URI.
 %End
 
     static QString removePassword( const QString &aUri );
 %Docstring
-Removes password element from uris
+Removes the password element a URI.
```
Comment (DelazJ) [inline python/core/auto_generated/qgsdatasourceuri.sip.in]: Missing word?
```suggestion
Removes the password element from a URI.
```

File: python/core/auto_generated/qgsdatasourceuri.sip.in [hunk 2]
```diff
@@ -139,144 +145,166 @@ Sets database
...
     QString geometryColumn() const;
 %Docstring
-Returns the name of the geometry column
+Returns the name of the geometry column stored in the URI, if set.
 %End
 
     void setUseEstimatedMetadata( bool flag );
 %Docstring
-Sets use Estimated Metadata
+Sets use whether estimated metadata should be used for the connection.
```
Comment (DelazJ) [inline python/core/auto_generated/qgsdatasourceuri.sip.in]: Should be ?
```suggestion
Sets whether estimated metadata should be used for the connection.
```

File: python/core/auto_generated/qgsdatasourceuri.sip.in [hunk 3]
```diff
@@ -31,52 +37,52 @@ class QgsDataSourceUri
...
     QgsDataSourceUri();
-%Docstring
-default constructor
-%End
 
-    QgsDataSourceUri( QString uri );
+    QgsDataSourceUri( const QString &uri );
 %Docstring
-constructor which parses input URI
+Constructor for QgsDataSourceUri which parses input an input ``uri`` string.
```
Comment (DelazJ) [inline python/core/auto_generated/qgsdatasourceuri.sip.in]:
```suggestion
Constructor for QgsDataSourceUri which parses an input ``uri`` string.
```

**Comment by nyalldawson** [comment]
Thanks @delazj.. I must've been half asleep when I wrote this :p

**Pull request merged**